lua/starfall/libs_sh/darkrp.lua

```diff
@@ -0,0 +1,1119 @@
+	moneyrequestBurst = SF.BurstObject("moneyrequest", "money request", 0.5, 1, "The rate at which the cooldown for money requests that can be made for a single player decreases per second. Lower is longer, higher is shorter.", "Number of money requests that can be made by a single player in a short time.")
+	
+	registerprivilege("darkrp.moneyPrinterHooks", "Get own money printer info", "Allows the user to know when their own money printers catch fire or print money (and how much was printed)")
+	registerprivilege("darkrp.playerWalletChanged", "Be notified of wallet changes", "Allows the user to know when their own wallet changes")
+	registerprivilege("darkrp.lockdownHooks", "Know when lockdowns begin and end", "Allows the user to know when a lockdown begins or ends")
+	registerprivilege("darkrp.lawHooks", "Know when laws change", "Allows the user to know when a law is added or removed, and when the laws are reset")
+	registerprivilege("darkrp.lockpickHooks", "Know when they start picking a lock", "Allows the user to know when they start or finish lockpicking")
+	registerprivilege("darkrp.requestMoney", "Ask players for money", "Allows the user to prompt other users for money (similar to E2 moneyRequest)")
+	registerprivilege("darkrp.giveMoney", "Give players money", "Allows the user to give other users money")
+	
+	MoneyRequest.requests = SF.EntityTable("MoneyRequests")
+
+	function MoneyRequest.Request:accept()
+		local sender, receiver, amount, instance = self.sender, self.receiver, self.amount, self.instance
+
+		if sender:canAfford(amount) then
+			printDebug("SF: Accepted money request.", self)
+			DarkRP.payPlayer(sender, receiver, amount)
+			if self.callbackSuccess then
+				instance:runFunction(self.callbackSuccess, self.message, instance.Types.Player.Wrap(sender), amount)
+			end
+		else
+			printDebug("SF: Attempted to accept money request but the sender couldn't afford it.", self)
+		end
+	end
+	
+	function MoneyRequest.Request:decline()
+		printDebug("SF: Declined money request.", self)
+		if self.callbackFailure then
+			self.instance:runFunction(self.callbackFailure, "REQUEST_DENIED")
+		end
+	end
+		
+	function MoneyRequest.Request:send()
+		net.Start("sf_moneyrequest2")
+			net.WriteEntity(self.receiver)
+			net.WriteUInt(self.amount, 32)
+			net.WriteFloat(self.expiry)
+			if self.message then
+				net.WriteUInt(#self.message, 8)
+				net.WriteData(self.message)
+			else
+				net.WriteUInt(0, 8)
+			end
+		net.Send(self.sender)
+	end
+
+	function MoneyRequest:exists(sender, receiver)
+		return self.requests[sender]~=nil and self.requests[sender][receiver]~=nil
+	end
+
+	function MoneyRequest:add(sender, receiver, amount, message, instance, callbackSuccess, callbackFailure)
+		if next(self.requests) == nil then hook.Add("Think","SF_DarkRpMoneyRequests",function() self:think() end) end
```

Comment:
Why is `MoneyRequest` an object when code like this means there will only ever be one of it?

Reply:
It's just cleaner. Technically we can change all `MoneyRequest:` to `MoneyRequest.` but I like using 'self' in the functions.